### PR TITLE
Issue #12 : Feature request, support for multiple repositories

### PR DESCRIPTION
**Pull request by dcilia**
Implemented the feature of having multiple repositories in Buildasaur

When launching the app you will now see a TableView of the projects configured, with the ability to:
- Create a new project/ add new project
- Configure a current project
- Delete a project by right clicking and selecting delete
- Double click a row and you will be taken to configure your project

Testing:

I tested this using Yosemite and beta Xcode Server 5.0.  I was not actually able to get Buildasaur to work on El Capitan with server 5.0

Architecture:
- Changed the storyboard to make the table view controller the initial view window
- added multiple segues
- minimal change to the StatusProjectViewController to make sure it now displayed data from the table view controller.
- Changed the storage manager to post a notification whenever the projects array is set.
- Added a protocol / datasource pattern for the tableview controller and project view controller to use the same data source.

### CODE REVIEW

**Comment by czechboy0** [comment]
Hi @dcilia,

this is amazing, thank you so much for taking the time! I'll build it myself locally and give it a try myself and I'll add comments to the code, I just quickly skimmed it! 

What was the problem with getting Buildasaur working on El Cap?

**Comment by czechboy0** [comment]
Also - what Xcode version were you building with? It'd be great to have it compatible with the latest Xcode 7 Beta (6), because that's what we're building towards. Xcode 6 compatible Buildasaur was versions 0.2.9 and earlier.

**Comment by dcilia** [comment]
@czechboy0 hey, no worries!

So I forgot to mention in my original request that I am still working on the "syncer" in that when you start the syncer with multiple projects open it defaults to first.  I thought I would throw this your way for now.

Happy to make any changes you need me to.  Let me know

I built this with Yosemite and Xcode Beta 6 (the latest).

**Comment by dcilia** [comment]
@czechboy0 don't recall the error on El Capitan offhand, I'll take a look again and get back to you.

**Comment by czechboy0** [comment]
Cool - Xcode 7 Beta 6 (I know you meant this, just clarifying for any future readers - we don't support Xcode 6 any more because of new Swift).

Sure feel free to ask questions, but also you can continue working on this (I'm a bit busy these couple of weeks) - even connecting the syncers. Thanks again for contributing back!

**Comment by dcilia** [comment]
@czechboy0 yep, just confirming Xcode version: Version 7.0 beta 6 (7A192o)

**Comment by czechboy0** [comment]
:+1:

**Comment by czechboy0** [comment]
Result of Integration **1**
---
*Duration*: 1 minute and 40 seconds
*Result*: **1 error, failing state: build-errors**

File: Buildasaur.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -84,6 +84,9 @@
 		3AD338B01AAE31D500ECD0F2 /* BuildTemplateViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3AD338AF1AAE31D500ECD0F2 /* BuildTemplateViewController.swift */; };
 		3AF090B81B1134AA0058567F /* BranchWatchingViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3AF090B71B1134AA0058567F /* BranchWatchingViewController.swift */; };
 		3AF1B1241AAC7CA500917EF3 /* StatusSyncerViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3AF1B1231AAC7CA500917EF3 /* StatusSyncerViewController.swift */; };
+		55A43BAC1B8CDE4300EED079 /* ProjectViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 55A43BAB1B8CDE4300EED079 /* ProjectViewController.swift */; settings = {ASSET_TAGS = (); }; };
+		6604F68F1B8F4E4900E70BA2 /* XcodeServerSDK.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6604F68E1B8F4E4900E70BA2 /* XcodeServerSDK.framework */; };
+		6604F6901B8F4E4900E70BA2 /* XcodeServerSDK.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = 6604F68E1B8F4E4900E70BA2 /* XcodeServerSDK.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Comment (czechboy0) [inline Buildasaur.xcodeproj/project.pbxproj:89]: Not really sure about these project changes - doesn't build on my Xcode Server - why is this added? We already have `XcodeServerSDK` added as a dependency with CocoaPods.

Comment (dcilia) [inline Buildasaur.xcodeproj/project.pbxproj:89]: hmmm, probably because I couldn't get it to compile, even after running Pod install.  Only until I manually added did it compile

Comment (czechboy0) [inline Buildasaur.xcodeproj/project.pbxproj:89]: Try to clone it cleanly in a new folder and just run `pod install` - that should be all you need to get it working (also make sure to merge `gladiolus` from my original repo, maybe you're behind on the `Podfile` dependencies?)

Comment (dcilia) [inline Buildasaur.xcodeproj/project.pbxproj:89]: Okay will give it try.

**Comment by czechboy0** [comment]
Please reopen this PR against `master`. Thanks! :+1:
